src/components/containers/Main.js

```diff
@@ -0,0 +1,47 @@
+import React, { Component } from 'react';
+import UrlInputForm from '../form/UrlInputForm';
+import Results from '../Results';
+import apiCall from '../../services/apiCall';
+
+export default class Main extends Component {
+
+  state = {
+    urlInput: '',
+    method: 'get',
+    inputJson: '',
+    resultsJson: ''
+  };
+
+  handleUrlChange = ({ target }) => {
+    this.setState({ urlInput: target.value });
+  }
+
+  handleMethodChange = ({ target }) => {
+    this.setState({ method: target.value });
+  }
+
+  handleInputJsonChange = ({ target }) => {
+    this.setState({ inputJson: target.value });
+  }
```

Comment:
handleMethodChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  }

yours are totally fine! but if you dont want to write a bunch of them, you can use this for all of them as long as your input tags have name attributes on them that match up with your state object.